cluster/manifests/03-ebs-csi/02-csi-driver.yaml

```diff
@@ -0,0 +1,127 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: csi-node-driver
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      application: container-storage-interface
+      component: driver
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        application: container-storage-interface
+        component: driver
+    spec:
+      containers:
+        - args:
+            - node
+            - --endpoint=$(CSI_ENDPOINT)
+            - --logtostderr
+            - --v=5
+          env:
+            - name: CSI_ENDPOINT
+              value: unix:/csi/csi.sock
+          image: registry.opensource.zalan.do/teapot/ebs-csi-driver:v0.5.0-master-2
+          livenessProbe:
+            failureThreshold: 5
+            httpGet:
+              path: /healthz
+              port: healthz
+            initialDelaySeconds: 10
+            periodSeconds: 10
+            timeoutSeconds: 3
+          name: ebs-plugin
+          ports:
+            - containerPort: 9808
+              name: healthz
+              protocol: TCP
+          resources:
+            requests:
+              cpu: 50m
+              memory: 50Mi
+            limits:
+              cpu: 50m
+              memory: 50Mi
+          securityContext:
+            privileged: true
+          volumeMounts:
+            - mountPath: /opt/podruntime/kubelet
+              mountPropagation: Bidirectional
+              name: kubelet-dir
+            - mountPath: /csi
+              name: plugin-dir
+            - mountPath: /dev
+              name: device-dir
+        - args:
+            - --csi-address=$(ADDRESS)
+            - --kubelet-registration-path=$(DRIVER_REG_SOCK_PATH)
+            - --v=5
+          env:
+            - name: ADDRESS
+              value: /csi/csi.sock
+            - name: DRIVER_REG_SOCK_PATH
+              value: /opt/podruntime/kubelet/plugins/ebs.csi.aws.com/csi.sock
+          image: registry.opensource.zalan.do/teapot/csi-node-driver-registrar:v1.1.0-master-1
+          lifecycle:
+            preStop:
+              exec:
+                command:
+                  - /bin/sh
+                  - -c
+                  - rm -rf /registration/ebs.csi.aws.com-reg.sock /csi/csi.sock
```

Comment:
See above, if we only cleanup in `preStop` it'll be in a broken state after a node reboot.

Reply:
What if we change it to a `preStart` hook?

Reply:
That should work. We can also use an initcontainer for this.

Reply:
I will change it to an `initContainer`. Also the socket file needs to be that particular directory for kubelet to talk to it.